pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -1127,6 +1127,7 @@ private void closePartitionMetadataProvider() {
    */
   private void cleanupMetrics() {
     _serverMetrics.removeTableGauge(_clientId, ServerGauge.LLC_PARTITION_CONSUMING);
+    _realtimeTableDataManager.onConsumingToDropped(_segmentNameStr);
```

Comment:
This is not just called when doing a rebalance. It for example is also called when segments are committed. In these cases, given we continue ingesting from them, the metric is created again some ms later. The same was happening with LLC_PARTITION_CONSUMING, so I assume this is a correct behavior

Reply:
Thanks for fixing this.

One corner case I can think of: if the creation of a new consuming segment gets stuck, then we miss both metrics. For LLC_PARTITION_CONSUMING, we use it to check if consuming is stopped or not, it is expected to miss it. The question is: do we want to stop emitting ingestion delay metric in this case? Is this a valid concern?

Reply:
The logic of the ingestion alert should be to be fired when:

```
For any partition 
  - sum LLC_PARTITION_CONSUMING by server = 0 
  - max(REALTIME_INGESTION_DELAY_MS) is higher than a value
```

In case the ingestion gets stuck after calling `cleanupMetrics` the first condition should be true and therefore the alert should be fired.

Reply:
@gortiz are we handling no metrics properly? (i.e. if there's no metric, we should treat it as 0?)

Reply:
Given we have some reticence on merging this change, I've been thinking we could solve our current false alert problem by multiplying REALTIME_INGESTION_DELAY_MS by LLC_PARTITION_CONSUMING (which is either 0 or 1).

Reply:
BTW, in case we don't want to call `onConsumingToDropped` here we should change the javadoc of this method, which is quite clear:

```java
  /**
   * Cleans up the metrics that reflects the state of the realtime segment.
   * This step is essential as the instance may not be the target location for some of the partitions.
   * E.g. if the number of partitions increases, or a host swap is needed, the target location for some partitions
   * may change,
   * and the current host remains to run. In this case, the current server would still keep the state of the old
   * partitions,
   * which no longer resides in this host any more, thus causes false positive information to the metric system.
   */
```

Reply:
> @gortiz are we handling no metrics properly? (i.e. if there's no metric, we should treat it as 0 on the alerting system?)

We are using Prometheus. In Prometheus 0 is different than non existent. When a metric is not existent, the expression evaluates to no value. That is consistent for example to what happens when a table is deleted.